### ISSUE TITLE
WPF - IContextMenuHandler callback doesn't execute when called on `WPF UI` thread.

### ISSUE THREAD

**Issue by sentialx** [cefsharp/CefSharp#1767]
Hi, I'm using Cefsharp 51 x86 on Windows 10 and I have problem with IContextMenuHandler. I created my custom context menu using this code: https://github.com/cefsharp/CefSharp/blob/e3c0d8b013d85dc41d4e01a573dd50506943fc7e/CefSharp.Wpf.Example/Handlers/MenuHandler.cs

```
if (commandId == (CefMenuCommand)26501)
            {
                browser.GetHost().ShowDevTools();
                return true;
            }
            if (commandId == (CefMenuCommand)26502)
            {
                browser.GetHost().CloseDevTools();
                return true;
            }
```

theese conditions are not firing when I click on the context menu item 26501 or 26502. I tried to add Console.WriteLine("something"); but still does not work. Before update to CefSharp 51 context menu was working fine

**Comment by amaitland** [comment]
Have you verified the problem exists when running using the `CefSharp.Wpf.Example`? If there is a problem, it's likely a bug in `CEF`.

**Comment by Sential** [comment]
Sorry I can't start CefSharp.Wpf.Example because throws me a lot of errors:
`
Error	CS0103	The name 'Cef' does not exist in the current context	CefSharp.WinForms.Example
`

**Comment by amaitland** [comment]
> If there is a problem, it's likely a bug in CEF.

@Sential `CefSharp` is just a wrapper, things break all the time when upgrading to a newer version of `CEF`.

It's important to note the build in `Context Menu` now works in `WPF`, so you can customize the built in one successfully as a workaround.

**Comment by amaitland** [comment]
From what I can tell this is a bug in `CEF` and should be reported on http://magpcss.org/ceforum/index.php

**Comment by amaitland** [comment]
It's unlikely the `upstream` issue will be resolved so `PR` #3815 will added a `native WPF` menu that can be used and extended.

**Comment by amaitland** [comment]
Starting in version `96` we'll use our custom `WPF` menu by default.
